vulkaninfo/vulkaninfo.c

```diff
@@ -2008,24 +2008,172 @@ bool FormatRangeSupported(const struct FormatRange *format_range, const struct A
     return false;
 }
 
+bool FormatPropsEq(const VkFormatProperties *props1, const VkFormatProperties *props2) {
+    if( props1->bufferFeatures == props2->bufferFeatures
+     && props1->linearTilingFeatures == props2->linearTilingFeatures
+     && props1->optimalTilingFeatures == props2->optimalTilingFeatures) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+struct PropFormats {
+    VkFormatProperties props;
+
+    uint32_t format_count;
+    uint32_t format_reserve;
+    VkFormat* formats;
+};
+
+void FormatPropsShortenedDump(const struct AppGpu *gpu) {
+    const VkFormatProperties unsupported_prop = {0};
+    uint32_t unique_props_count = 1;
+    uint32_t unique_props_reserve = 50;
+    struct PropFormats *prop_map = malloc(sizeof(struct PropFormats) * unique_props_reserve);
+    if (!prop_map) ERR_EXIT(VK_ERROR_OUT_OF_HOST_MEMORY);
+    prop_map[0].props = unsupported_prop;
+    prop_map[0].format_count = 0;
+    prop_map[0].format_reserve = 20;
+    prop_map[0].formats = malloc(sizeof(VkFormat) * prop_map[0].format_reserve);
+    if (!prop_map[0].formats) ERR_EXIT(VK_ERROR_OUT_OF_HOST_MEMORY);
+
+    for (uint32_t ri = 0; ri < ARRAY_SIZE(supported_format_ranges); ++ri) {
+        struct FormatRange format_range = supported_format_ranges[ri];
+        if (FormatRangeSupported(&format_range, gpu)) {
+            for (VkFormat fmt = format_range.first_format; fmt <= format_range.last_format; ++fmt) {
+                VkFormatProperties props;
+                vkGetPhysicalDeviceFormatProperties(gpu->obj, fmt, &props);
+
+                uint32_t formats_prop_i = 0;
+                for (; formats_prop_i < unique_props_count; ++formats_prop_i) {
+                     if( FormatPropsEq(&prop_map[formats_prop_i].props, &props) ) break;
+                }
+
+                if( formats_prop_i < unique_props_count ){
+                    struct PropFormats *propFormats = &prop_map[formats_prop_i];
+                    ++propFormats->format_count;
+
+                    if( propFormats->format_count > propFormats->format_reserve ){
+                        propFormats->format_reserve *= 2;
+                        propFormats->formats = realloc( propFormats->formats, sizeof(VkFormat) * propFormats->format_reserve);
+                        if (!propFormats->formats) ERR_EXIT(VK_ERROR_OUT_OF_HOST_MEMORY);
+                    }
+
+                    propFormats->formats[propFormats->format_count - 1] = fmt;
+                }
+                else {
```

Comment:
The checks will get mad at this do to improper formatting.  Make sure you run clang-format on your changes.

Reply:
I did, but I think something is wrong with it as it did no changes :/

Reply:
Just tried it on mine and no changes either.  Well, let's see what CI says and if it's good then who am I to argue.

Reply:
Google is C++ style, no? Maybe it just skips C files?

Reply:
Oh, there's `clang-format off` macro in the code :D